utils/resultstable.go

```diff
@@ -352,36 +354,43 @@ func prepareSecrets(secrets []*sarif.Run, isTable bool) []formats.SourceCodeRow
 				currSeverity = severityutils.Unknown
 			}
 			for _, location := range secretResult.Locations {
-				secretsRows = append(secretsRows,
-					formats.SourceCodeRow{
-						SeverityDetails: severityutils.GetAsDetails(currSeverity, jasutils.Applicable, isTable),
-						Finding:         sarifutils.GetResultMsgText(secretResult),
-						Location: formats.Location{
-							File:        sarifutils.GetRelativeLocationFileName(location, secretRun.Invocations),
-							StartLine:   sarifutils.GetLocationStartLine(location),
-							StartColumn: sarifutils.GetLocationStartColumn(location),
-							EndLine:     sarifutils.GetLocationEndLine(location),
-							EndColumn:   sarifutils.GetLocationEndColumn(location),
-							Snippet:     sarifutils.GetLocationSnippet(location),
-						},
+				newRow := formats.SourceCodeRow{
+					SeverityDetails: severityutils.GetAsDetails(currSeverity, jasutils.Applicable, isTable),
+					Finding:         sarifutils.GetResultMsgText(secretResult),
+					Location: formats.Location{
+						File:        sarifutils.GetRelativeLocationFileName(location, secretRun.Invocations),
+						StartLine:   sarifutils.GetLocationStartLine(location),
+						StartColumn: sarifutils.GetLocationStartColumn(location),
+						EndLine:     sarifutils.GetLocationEndLine(location),
+						EndColumn:   sarifutils.GetLocationEndColumn(location),
+						Snippet:     sarifutils.GetLocationSnippet(location),
 					},
-				)
+				}
+				if tokenValidation := sarifutils.GetResultPropertyTokenValidation(secretResult); tokenValidation != "" {
+					newRow.TokenValidation = sarifutils.GetResultPropertyTokenValidation(secretResult)
+					newRow.TokenInfo = sarifutils.GetResultPropertyMetadata(secretResult)
+					tokenValidationActivated = true
+				}
+				secretsRows = append(secretsRows, newRow)
 			}
 		}
 	}
 
 	sort.Slice(secretsRows, func(i, j int) bool {
+		if jasutils.TokenValidationOrder[secretsRows[i].TokenValidation] != jasutils.TokenValidationOrder[secretsRows[j].TokenValidation] {
+			return jasutils.TokenValidationOrder[secretsRows[i].TokenValidation] < jasutils.TokenValidationOrder[secretsRows[j].TokenValidation]
+		}
 		return secretsRows[i].SeverityNumValue > secretsRows[j].SeverityNumValue
```

Comment:
I think you need more complex sorting like with "ApplicableStatus" + Severity.
If the status is `Active` or `Unsupported` or `Not scanned` it should be base on severity and above the `not active tokens`.
Right now the order does not make sense

Reply:
add tests for the sorting as well

Reply:
we discussed about this, also added tests for the sorting